pallets/stellar-bridge/src/lib.rs

```diff
@@ -332,7 +346,7 @@ pub mod pallet {
                         debug::info!("✴️  New transaction from Horizon (id {:#?}). Starting to replicate transaction in Pendulum.", str::from_utf8(&saved_tx_id).unwrap());
 
                         let mut amount: Option<BalanceOf<T>> = None;
-                        let destination = T::GatewayMockedDestination::get();
+                        let mut destination: Option<T::AccountId> = None;
                         let mut currency = None;
```

Comment:
Instead of
```
let mut a: Option<X> = None;
let mut b: Option<Y> = None;

if ... {...}
```

Just make
```
let (a,b) = if ... {...}
```

Reply:
You could break the code here up into something like (I didn't check whether it compiles, maybe some adaptations required).

```rust
        fn process_new_transaction(transaction: stellar::types::Transaction) {
            let destination =
                if let stellar::MuxedAccount::KeyTypeEd25519(key) = transaction.source_account {
                    let pubkey = stellar::PublicKey::from_binary(key);
                    Some(T::AddressConversion::unlookup(pubkey))
                } else {
                    debug::error!("❌  Source account format not supported.");
                    return;
                };

            for op in transaction.operations.get_vec() {
                if let stellar::types::OperationBody::Payment(payment_op) = &op.body {
                    let dest_account = stellar::MuxedAccount::from(payment_op.destination.clone());
                    debug::info!("Muxed account {:#?}", dest_account);

                    if let stellar::MuxedAccount::KeyTypeEd25519(payment_dest_public_key) =
                        payment_op.destination
                    {
                        if Self::is_escrow(payment_dest_public_key) {
                            let amount = T::BalanceConversion::unlookup(payment_op.amount);
                            let currency =
                                T::CurrencyConversion::unlookup(payment_op.asset.clone());

                            debug::info!("Pendulum address for deposit {:?}", destination);
                            debug::info!("Currency {:?}", currency);
                            debug::info!("Amount {:?}", amount);

                            match Self::offchain_unsigned_tx_signed_payload(
                                currency,
                                amount,
                                destination,
                            ) {
                                Err(_) => debug::warn!("Sending the tx failed."),
                                Ok(_) => (),
                            }
                        }
                    }
                }
            }
        }

        fn handle_new_transaction(tx: &Transaction) {
            const UP_TO_DATE: () = ();

            let latest_tx_id_utf8 = &tx.id;

            let id_storage = StorageValueRef::persistent(b"stellar-watch:last-tx-id");

            let prev_read = id_storage.get::<Vec<u8>>();
            let initial = !matches!(prev_read, Some(Some(_)));

            let res = id_storage.mutate(|last_stored_tx_id: Option<Option<Vec<u8>>>| {
                match last_stored_tx_id {
                    Some(Some(value)) if value == *latest_tx_id_utf8 => Err(UP_TO_DATE),
                    _ => Ok(latest_tx_id_utf8.clone()),
                }
            });

            match res {
                Ok(Ok(saved_tx_id)) => {
                    if !initial {
                        debug::info!("✴️  New transaction from Horizon (id {:#?}). Starting to replicate transaction in Pendulum.", str::from_utf8(&saved_tx_id).unwrap());

                        // Decode transaction to Base64 and then to Stellar XDR to get transaction details
                        let tx_xdr = base64::decode(&tx.envelope_xdr).unwrap();
                        let tx_envelope = stellar::TransactionEnvelope::from_xdr(&tx_xdr).unwrap();

                        if let stellar::TransactionEnvelope::EnvelopeTypeTx(env) = tx_envelope {
                            Self::process_new_transaction(env.tx);
                        }
                    }
                }
                Err(UP_TO_DATE) => {
                    debug::info!("Already up to date");
                }
                Ok(Err(_)) => {
                    debug::info!("Failed to save last transaction id.");
                }
            }
        }
    }
```

Reply:
The nested `if`s are ugly and can maybe be avoided by something like:

```
let payment_op = if let stellar::types::OperationBody::Payment(payment_op) = &op.body {payment_op} else {continue;}

...
```

Reply:
To took your example as inspiration, but replaced the nested if with a `filter_map()` over the array of operations, so we only get `PaymentOp`s directly.